### PR TITLE
`<SortableList/>` - multiple columns bug fix

### PR DESCRIPTION
**Pull request by frontned**
### 🔦 Summary

Fixing SortableList multiple columns dragging outside case.

### CODE REVIEW

File: src/DragAndDrop/Draggable/components/DraggableSource.js [hunk 1]
```diff
@@ -62,23 +75,49 @@ const source = {
...
+        originalItem,
+        removedIndex,
+        addedIndex,
+        addedToContainerId,
+        removedFromContainerId,
+      } = dragObject;
+
+      if (sameGroup) {
+        onDrop({
+          payload: originalItem, // original item
```
Comment (argshook) [inline src/DragAndDrop/Draggable/components/DraggableSource.js]: remove comments that serve no purpose

File: src/DragAndDrop/Draggable/components/DraggableSource.js [hunk 2]
```diff
@@ -62,23 +75,49 @@ const source = {
...
+          addedToContainerId: dropResult.containerId, // new container for item
           removedFromContainerId: containerId, // original item container
         });
       }
+    } else {
+      const isSameContainer =
+        dragObject && containerId === dragObject.addedToContainerId;
+      const sameGroup = isSameGroup(dragObject, groupName);
+
+      // If item is dragged anywhere but other container
```
Comment (argshook) [inline src/DragAndDrop/Draggable/components/DraggableSource.js]: comment seems to contradict code?

File: src/DragAndDrop/Draggable/components/DraggableSource.js [hunk 3]
```diff
@@ -10,6 +10,19 @@ import { ItemTypes } from '../types';
 
 /* eslint-disable new-cap */
 
+// monitor.dropResult() returns null when item is dropped outside the drop area so we need to track everything by ourselves
```
Comment (argshook) [inline src/DragAndDrop/Draggable/components/DraggableSource.js]: you mean `monitor.getDropResult()`?

File: src/DragAndDrop/Draggable/components/DraggableSource.js [hunk 4]
```diff
@@ -62,23 +75,49 @@ const source = {
...
+          payload: dropItem.originalItem, // original item
+          removedIndex: dropItem.originalIndex, // original item index
+          addedIndex: dropItem.index, // new item index
+          addedToContainerId: dropResult.containerId, // new container for item
           removedFromContainerId: containerId, // original item container
         });
       }
+    } else {
+      const isSameContainer =
+        dragObject && containerId === dragObject.addedToContainerId;
```
Comment (argshook) [inline src/DragAndDrop/Draggable/components/DraggableSource.js]:
```suggestion
        containerId === dragObject.addedToContainerId;
```

`dragObject` seems to always be defined (truthy)

File: src/DragAndDrop/Draggable/components/DraggableSource.js [hunk 5]
```diff
@@ -94,17 +133,25 @@ const source = {
...
+    return (isSameGroup(item, groupName) || isSameContainer) && item.id === id;
   },
 };
 
 const collect = (connect, monitor) => ({
   connectDragSource: connect.dragSource(),
   connectDragPreview: connect.dragPreview(),
   isDragging: monitor.isDragging(),
+  // Need to set added index from component, since isDragging index is one step behind
+  setAddedIndex: index => (dragObject.addedIndex = index),
```
Comment (argshook) [inline src/DragAndDrop/Draggable/components/DraggableSource.js]:
```suggestion
  setAddedIndex: index => { dragObject.addedIndex = index },
```

let's do regular code block curly brackets, braces might trip over other readers

**Comment by frontned** [comment]
Safari has issue with multiple hover states. On Multi Area List with sortable columns.

**Comment by TalZa1810** [comment]
Hi @argshook ,
Can you handle this PR?

**Comment by argshook** [comment]
opened separate issue describing the bug. This PR tries to fix it, however it creates another issue on safari.

closing for now, this should be tracked on separate issue. This PR should be revisited.

https://github.com/wix/wix-style-react/issues/4804

**Pull request closed without merging**